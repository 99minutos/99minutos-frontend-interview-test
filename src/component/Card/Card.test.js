import { render, screen } from "@testing-library/react"
import CardSpace from './index';

describe('CardSpace', ()=>{

    const data = {
        id:1,
        mission_name: 'test',
        details: 'lorem ipsum',
        links:{flickr_image:''},
    }

    let card = null;

    it('should render empty data', ()=>{
        render(<CardSpace space={{}}/>)
        expect(screen.getByText(/No Data Available/i)).toBeInTheDocument();
    })
    it('should render card with data', ()=>{
        card = render(<CardSpace 
            space={data}
            />);
        expect(screen.getByText(`${data.mission_name}`)).toBeInTheDocument();
        expect(screen.getByText(`${data.details}`)).toBeInTheDocument();
        expect(card).toMatchSnapshot()
    });
})