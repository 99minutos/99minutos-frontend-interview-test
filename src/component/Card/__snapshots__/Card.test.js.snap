// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSpace should render card with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardMedia-root img-container"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular makeStyles-large-2 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </div>
        <div
          class="card-details"
        >
          <div
            class="MuiCardContent-root"
          >
            <h5
              class="title"
            >
              test
            </h5>
            <p
              class="description"
            >
              lorem ipsum
            </p>
            <p
              class="date"
            >
              No Data Available
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardMedia-root img-container"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular makeStyles-large-2 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiAvatar-fallback"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
      <div
        class="card-details"
      >
        <div
          class="MuiCardContent-root"
        >
          <h5
            class="title"
          >
            test
          </h5>
          <p
            class="description"
          >
            lorem ipsum
          </p>
          <p
            class="date"
          >
            No Data Available
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
